### PR TITLE
CI: remove duplicated code check #36642

### LINKED ISSUE #36642: CI: Deduplicate linting checks
https://github.com/pandas-dev/pandas/pull/36471 added some linting checks (e.g., black, flake8) to a pre-commit workflow which are duplicated in https://github.com/pandas-dev/pandas/blob/master/ci/code_checks.sh. Once we feel good that the pre-commit checks are right we should go ahead and delete them from `code_checks.sh`.

cc @MarcoGorelli

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #36642

### CODE REVIEW

**Comment by dsaxton** [comment]
I wonder if it's also possible to move flake8-rst into pre-commit, perhaps as a local hook. Maybe @MarcoGorelli knows?

**Comment by fangchenli** [comment]
xref #34150 flake8-rst is not under active development anymore.

**Comment by MarcoGorelli** [comment]
Is flake8-rst the only thing blocking us from upgrading flake8? If so, I'd say let's fork it, run it in pre-commit, and no longer pin flake8

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -37,12 +37,6 @@ jobs:
         ci/code_checks.sh lint
       if: always()
 
-    - name: Dependencies consistency
-      run: |
-        source activate pandas-dev
-        ci/code_checks.sh dependencies
```
Comment (WillAyd) [inline .github/workflows/ci.yml:43]: Do we check for `dependencies` now anywhere? If not should just remove from code_checks.sh altogether too

**Review by WillAyd — COMMENTED** [review]
Looks good. Comment on potential additional cleanup otherwise lgtm

**Comment by WillAyd** [comment]
@simonjayhawkins I think should back port?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fangchenli) [reply .github/workflows/ci.yml:43]: It's checked in pre-commit.

Comment (WillAyd) [reply .github/workflows/ci.yml:43]: Clarification - I am asking if the `dependencies` argument is used anywhere. I don't think it is but has relics of it in code_checks (?)

**Review by WillAyd — APPROVED** [review]
lgtm

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fangchenli) [reply .github/workflows/ci.yml:43]: Sorry... misunderstanding. It's not used anywhere else. And I just removed related lines from code_check.sh.

**Comment by MarcoGorelli** [comment]
Sorry to backtrack, but TBH I'd appreciate if we could wait a little before merging this.

I noticed in #36978 that isort, when run in pre-commit, wasn't picking up `.pxd` files (which the newest version of isort supports), even though they're picked up when running `isort` from the terminal. So first, I think that a version of #36978 which passes both code_checks and pre-commit should be in

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins I think should back port?

I _think_ better to keep the backport without the newer style checks. If we release 1.2 in November, 1.1.4 maybe the last in the 1.1.x Series. so maybe not that many more regressions to backport now anyhow.

**Comment by MarcoGorelli** [comment]
Hi @fangchenli 

We now have two separate hooks for isort:
```
    -   id: isort
        name: isort (python)
    -   id: isort
        name: isort (cython)
        types: [cython]
```
so IMO it's OK to remove it from `code_checks`. Can you fetch and merge upstream/master?

**Review by MarcoGorelli — APPROVED** [review]
Nice!

**Comment by WillAyd** [comment]
Can you merge master to get CI green?

**Comment by jreback** [comment]
thanks @fangchenli

**Pull request merged**